lib/DxcSupport/WinAdapter.cpp

```diff
@@ -62,21 +62,40 @@ unsigned int SysStringLen(const BSTR bstrString) {
 }
 //===---------------------- Char converstion ------------------------------===//
 
-const char *CPToLocale(uint32_t CodePage) {
+const char *SetLocaleForCodePage(int Category, uint32_t CodePage) {
 #ifdef __APPLE__
-  static const char *utf8 = "en_US.UTF-8";
-  static const char *iso88591 = "en_US.ISO8859-1";
+    switch(CodePage)
+    {
+      case CP_ACP:
+          return setlocale(Category, "en_US.ISO8859-1");
+      case CP_UTF8:
+          return setlocale(Category, "en_US.UTF-8");
+      default:
+          return nullptr;
+    }
 #else
-  static const char *utf8 = "en_US.utf8";
-  static const char *iso88591 = "en_US.iso88591";
+    const char* utf8LocaleOptions[] = {
+      "en_US.utf8", // supported on Ubuntu
+      "en_US.UTF-8" // supported on Mariner
```

Comment:
Is it right to talk about Ubuntu and Mariner here?  There are more distros than that available.

Reply:
I accidently resolved this conversation.

Reply:
I can remove these comments and just treat this list as possible options that work on different distros.  Nothing too specific.